src/Microsoft.Restier.Core/Conventions/ConventionBasedQueryExpressionProcessor.cs

```diff
@@ -149,7 +149,42 @@ private Expression AppendOnFilterExpression(QueryExpressionContext context, stri
                 var collectionType = context.VisitedNode.Type.FindGenericType(typeof(ICollection<>));
                 if (collectionType == null)
                 {
-                    return null;
+                    var nullableType = context.VisitedNode.Type;
```

Comment:
@robertmclaws  Because this part of code is responsible for applying filter on expanded single properties ;)
If you want to see changes in counting functionality check this commit: https://github.com/mateusz-malicki/RESTier/commit/7d724f14ec889e16191ae305c12624970cb643c6
btw, I'm sorry if i misused pull request in that case (commit https://github.com/mateusz-malicki/RESTier/commit/7344a9f4e82431406246f9c3cd002fb477a96562 has nothing to do about counting), that was my first pull request ever ;)